llvm/lib/CodeGen/MIRParser/MIParser.cpp

```diff
@@ -1175,19 +1175,10 @@ bool MIParser::parse(MachineInstr *&MI) {
   MI = MF.CreateMachineInstr(MCID, DebugLocation, /*NoImplicit=*/true);
   MI->setFlags(Flags);
 
-  unsigned NumExplicitOps = 0;
-  for (const auto &Operand : Operands) {
-    bool IsImplicitOp = Operand.Operand.isReg() && Operand.Operand.isImplicit();
-    if (!IsImplicitOp) {
-      if (!MCID.isVariadic() && NumExplicitOps >= MCID.getNumOperands() &&
-          !Operand.Operand.isValidExcessOperand())
-        return error(Operand.Begin, "too many operands for instruction");
-
-      ++NumExplicitOps;
-    }
-
+  // Don't check the operands make sense, let the verifier catch any
```

Comment:
I think there is "if" missing in the comment.

Reply:
It reads OK to me. "Don't check the operands make sense" is a shorter way of saying "Don't check _that_ the operands make sense".

Reply:
I somewhat misparsed it the first time I read it, and then it looked weird to me every time I looked at it.  :)